### PR TITLE
lib: nrf_modem_lib: update offloading of getaddrinfo errors

### PR DESCRIPTION
**Pull request by lemrey**
Update error table for getaddrinfo errors.
cc @Rallare

### CODE REVIEW

File: lib/nrf_modem_lib/nrf91_sockets.c [hunk 1]
```diff
@@ -485,21 +486,36 @@ static int z_to_nrf_protocol(int proto)
...
+	case NRF_EAI_SERVICE:
+		return DNS_EAI_SERVICE;
+	case NRF_EAI_SOCKTYPE:
+		return DNS_EAI_SOCKTYPE;
+	case NRF_EAI_INPROGRESS:
 		return DNS_EAI_INPROGRESS;
-	case NRF_ENETUNREACH:
-		errno = ENETUNREACH;
-	default:
+	case NRF_EAI_SYSTEM:
```
Comment (trantanen) [inline lib/nrf_modem_lib/nrf91_sockets.c:514]: Where will errno be set for EAI_SYSTEM?

Comment (tokangas) [reply lib/nrf_modem_lib/nrf91_sockets.c:514]: I think `errno` is already set by `nrf_getaddrinfo()` when it returns `NRF_EAI_SYSTEM`, so this should be enough?

**Review by tokangas — APPROVED** [review]
Looks good to me. 👍 

@lemrey BTW, the `nrf_getaddrinfo()` description in nrf_socket.h is a bit vague about the return values. It only states that the function returns an errno, but doesn't say that the values are from nrf_gai_errors.h, not nrf_errno.h. Maybe this should be clarified?

File: lib/nrf_modem_lib/nrf91_sockets.c (hunk 1 above)
Comment (rlubos) [reply lib/nrf_modem_lib/nrf91_sockets.c:514]: That's what I'd expect as well - @lemrey can you confirm that libmodem sets `errno` in this case?

**Review by rlubos — APPROVED** [review]

File: lib/nrf_modem_lib/nrf91_sockets.c (hunk 1 above)
Comment (lemrey) [reply lib/nrf_modem_lib/nrf91_sockets.c:514]: yep, `errno` is set by `libmodem` when `NRF_EAI_SYSTEM` is returned.

**Comment by lemrey** [comment]
Thanks @tokangas, we can clarify the doc in the header in the next release.

**Review by trantanen — APPROVED** [review]

**Pull request merged**